bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/discovery/DiscoveryResource.java

```diff
@@ -96,13 +116,59 @@ public Response getDiscoveryServices() {
         return Response.ok(new LinkedHashSet<>(supportedBindings)).build();
     }
 
+    @GET
+    @Path("/bindings/{bindingId}/info")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(operationId = "getDiscoveryServicesInfo", summary = "Gets information about the discovery services for a binding.", responses = {
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = DiscoveryInfoDTO.class))),
+            @ApiResponse(responseCode = "404", description = "Discovery service not found") })
+    public Response getDiscoveryServicesInfo(
+            @PathParam("bindingId") @Parameter(description = "binding Id") final String bindingId,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language) {
+        final Locale locale = localeService.getLocale(language);
+        String label = null;
+        String description = null;
+        boolean supported = false;
+        Set<DiscoveryService> discoveryServices = discoveryServiceRegistry.getDiscoveryServices(bindingId);
+
+        if (discoveryServices.isEmpty()) {
+            return JSONResponse.createResponse(Status.NOT_FOUND, null,
+                    "No discovery service found for binding " + bindingId);
```

Comment:
I am not sure if I should rather return `Response.status(Status.NOT_FOUND).build()`
I find the two approaches used in core framework without understanding which one is the most adapted.

Using `JSONResponse.createResponse` allows returning a JSON with an error message in addition to the simple HTTP status.

Reply:
I would prefer your current approach because it provides the additional info, which is more user-friendly, but TBH I don’t know if anyone really „reads“ that response.